WebAPI/Program.cs

```diff
@@ -1,8 +1,37 @@
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using AutoMapper;
+using Business.Mappings;
+using WebNetSample.Core.DependencyResolvers;
+using WebNetSample.Core.Extensions;
+using WebNetSample.Core.Utilities.IoC;
+using WebNetSample.DataAccess;
+using WebNetSample.WebAPI.DependencyResolvers;
+
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
 
+builder.Host.UseServiceProviderFactory(new AutofacServiceProviderFactory());
+builder.Host.ConfigureContainer<ContainerBuilder>(builder => builder.RegisterModule(new AutofacDependencyResolver()));
+
+builder.Services.AddDependencyResolvers(new ICoreModule[]
+{
+    new CoreModule()
+});
+
+var mapperConfig = new MapperConfiguration(config =>
+{
+    config.AddProfile(new MappingProfile());
+});
+
+IMapper mapper = mapperConfig.CreateMapper();
+builder.Services.AddSingleton(mapper);
```

Comment:
Let's move these configs to separate extension app configuration class to avoid making `Program.cs` too overloaded